### DISCUSSION TITLE
NX-Plugin for generating CODEOWNERS files with reviewers in project.json file

### DISCUSSION THREAD

**Discussion by swapniltech0390** [nrwl/nx#23352] [Feature Requests]
- [x] I'd be willing to implement this feature ([contributing guide](https://github.com/nrwl/nx/blob/master/CONTRIBUTING.md))

## Description
I would like to add a new NX plugin CODEOWNERS which will iterate through all apps and libraries and help to generate CODEOWNERS file for providing file/ folder level PR review access, which is Currenlty managed manually and it gets tedious to maintain large size projects on monorepo.

## Motivation
Currently, it is very difficult to manually write name of team members of application specific reviewers in **CODEOWNERS**. Specially when team changes frequently and list need to be updated periodically. 

## Suggested Implementation
We can add a new property in **project.json** file as **reviewers** list which will contain **GITHUB** id's of members of team which need to review the PR this will speed up the review process and streamline modifying the CODEOWNERS file without manual touch.

## Alternate Implementations
As of now the only solution is to manually do it in CODEOWNERS file. This is very tedious to maintain and modify the file for large projects

**Comment by jaysoo** [comment]
This is something we've discussed, but is not a priority currently. The best solution for now is to create a local plugin (install `@nx/plugin` then `nx g @nx/plugin:plugin my-plugin`) and add a generator with this capability.

**Comment by swapniltech0390** [comment]
@jaysoo  & @AgentEnder  Thank you for the quick response.
I have created a quick executor for this feature and released on node registry. Please spare sometime and share your feedbacks if possible.
npm i @swapniltech0390/nx-codeowners

**Comment by FrozenPandaz** [comment]
I've converted this to a discussion so it can be discussed more.

What kind of CODEOWNERS type of behavior would people like to see within Nx?
